ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRao.java

```diff
@@ -114,7 +114,9 @@ private void iterateOnTree() {
             updateOptimalLeafWithNextDepthBestLeaf();
             hasImproved = previousDepthOptimalLeaf != optimalLeaf; // It means this depth evaluation has improved the global cost
             if (hasImproved && previousDepthOptimalLeaf != rootLeaf) {
-                previousDepthOptimalLeaf.clearVariants();
+                previousDepthOptimalLeaf.clearAllVariants();
+            } else if (hasImproved) {
+                previousDepthOptimalLeaf.clearAllVariantsExceptInitialOne();
```

Comment:
This test is a bit redundant, maybe if(hasImproved) and then test on the leaves

Reply:
indeed! done